app/models/spree/product.rb

```diff
@@ -53,7 +53,7 @@ class Product < ApplicationRecord
     validates :name, presence: true
 
     validates :variant_unit, presence: true
-    validates :unit_value, presence:
+    validates :unit_value, numericality: { greater_than: 0 }, allow_nil: true, presence:
       { if: ->(p) { %w(weight volume).include?(p.variant_unit) && new_record? } }
```

Comment:
I find this quite confusing. We require it to be a number or nil but then it has to be present under certain conditions. What if we validate numericality under these conditions? I mean numericality instead of presence. Then we have fewer validators and we don't need to allow nil because it's automatically allowed when the condition isn't met.

We also got the same validation on the variant. We need to match the two, otherwise we'll have cases that are impossible to validate.